src/Rendering.md

```diff
@@ -1,85 +1,72 @@
+  1  | Waiting until the next frame               | 4560 dots (10 scanlines)             | VRAM, OAM, CGB palettes
 
-The background is composed of a _tilemap_. A tilemap is a
-large grid of tiles. However, tiles aren't directly written to tilemaps,
-they merely contain references to the tiles.
-This makes reusing tiles very cheap, both in CPU time and in
-required memory space, and it is the main mechanism that helps work around the
-paltry 8 KiB of video RAM.
+## Mode 3 length
 
-The background can be made to scroll as a whole, writing to two
-hardware registers. This makes scrolling very cheap.
+During Mode 3, by default the PPU outputs one pixel to the screen per dot; the screen is 160 pixels wide, so the minimum Mode 3 length is 160 + 12[^first12] = 172 dots.
 
-### Window
+Unlike most game consoles, the Game Boy does not always output pixels steadily[^crt]: some features cause the rendering process to stall for a couple dots.
+Any extra time spent stalling *lengthens* Mode 3; but since scanlines last for a fixed number of dots, Mode 0 is therefore shortened by that same amount of time.
 
-The window is sort of a second background layer on top of the background.
-It is fairly limited: it has no transparency, it's always a
-rectangle and only the position of the top-left pixel can be controlled.
+Three things can cause Mode 3 "penalties":
 
-Possible usage include a fixed status bar in an otherwise scrolling game (e.g.
-_Super Mario Land 2_).
+- **Background scrolling**: At the very beginning of Mode 3, rendering is paused for [`SCX`](<#FF42–FF43 — SCY, SCX: Viewport Y position, X position>) % 8 dots while the same number of pixels are discarded from the leftmost tile.
+- **Window**: After the last non-window pixel is emitted, a 6-dot penalty is incurred while the BG fetcher is being set up for the window.
+- **Objects**: Each object drawn during the scanline (even partially) incurs a 6- to 11-dot penalty ([see below](<#OBJ penalty algorithm>)).
 
-### Objects
+On DMG and GBC in DMG mode, mid-scanline writes to [`BGP`](<#FF47 — BGP (Non-CGB Mode only): BG palette data>) allow observing this behavior precisely, as any delay shifts the write's effect to the left by that many dots.
 
-The background layer is useful for elements scrolling as a whole, but
-it's impractical for objects that need to move separately, such as the player.
+### OBJ penalty algorithm
 
-The _objects_ layer is designed to fill this gap: _objects_ are made of 1 or 2 stacked tiles (8×8 or 8×16 pixels)
-and can be displayed anywhere on the screen.
+Only the OBJ's leftmost pixel matters here, transparent or not; it is designated as "The Pixel" in the following.
 
-::: tip NOTE
+1. Determine the tile (background or window) that The Pixel is within. (This is affected by horizontal scrolling and/or the window!)
+2. If that tile has **not** been considered by a previous OBJ yet:
+   1. Count how many of that tile's pixels are to the right of The Pixel.
```

Comment:
The wording "to the right" sounds exclusive, but I believe the logic here is inclusive.

Reply:
The subtraction below should have been adjusted to account for that.

Reply:
I guess it's not, because then the maximum is 7 pixels; minus 3 = 4; plus the "flat" 6 = 40, but the maximum stated above is 11. So probably that the line below should be "Subtract 2." instead?